golem-worker-service-base/src/worker_bridge_execution/content_type_mapper.rs

```diff
@@ -0,0 +1,900 @@
+        a: A,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError> {
+        let json = serde_json::to_value(&a).map_err(|_| {
+            ContentTypeMapError::internal(format!("Failed to serialise {} to json", a))
+        })?;
+        let body = Body::from_json(json)
+            .map_err(|_| ContentTypeMapError::internal("Failed to create body from JSON"))?;
+
+        Ok(body.with_content_type(ContentType::json().to_string()))
+    }
+
+    fn get_json_null() -> Result<WithContentType<Body>, ContentTypeMapError> {
+        Body::from_json(serde_json::Value::Null)
+            .map(|body| body.with_content_type(ContentType::json().to_string()))
+            .map_err(|_| ContentTypeMapError::internal("Failed to convert to json body"))
+    }
+
+    fn handle_complex<A: ContentTypeHeaderExt + Display>(
+        complex: &TypeAnnotatedValue,
+        content_header: &A,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError> {
+        if content_header.has_application_json() {
+            get_json(complex)
+        } else {
+            Err(ContentTypeMapError::illegal_mapping(
+                &AnalysedType::from(complex),
+                content_header,
+            ))
+        }
+    }
+
+    fn handle_list<A: ContentTypeHeaderExt + Display>(
+        original: &TypeAnnotatedValue,
+        inner_values: &[TypeAnnotatedValue],
+        list_type: &AnalysedType,
+        content_header: &A,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError> {
+        match list_type {
+            AnalysedType::U8 => {
+                let byte_stream = get_byte_stream(inner_values)?;
+                let body = Body::from_bytes(bytes::Bytes::from(byte_stream));
+                let content_type_header = content_header.response_content_type()?;
+                Ok(body.with_content_type(content_type_header.to_string()))
+            }
+            _ => {
+                if content_header.has_application_json() {
+                    get_json(original)
+                } else {
+                    Err(ContentTypeMapError::illegal_mapping(
+                        list_type,
+                        content_header,
+                    ))
+                }
+            }
+        }
+    }
+
+    fn handle_primitive<A: Display + serde::Serialize, B: ContentTypeHeaderExt + Display>(
+        input: &A,
+        primitive_type: &AnalysedType,
+        content_header: &B,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError> where {
+        if content_header.has_application_json() {
+            let json = serde_json::to_value(input)
+                .map_err(|_| ContentTypeMapError::internal("Failed to convert to json body"))?;
+
+            let body = Body::from_json(json).map_err(|_| {
+                ContentTypeMapError::internal(format!("Failed to convert {} to json body", input))
+            })?;
+
+            Ok(body.with_content_type(ContentType::json().to_string()))
+        } else {
```

Comment:
i think we should be able to support ContentType `text`, and just do `to_string`, for primitive types

Reply:
A previous commit had support for supporting text. But then decided to strictly follow the spec to avoid confusion that the PR doesn't have anything over what we discussed.


<img width="727" alt="image" src="https://github.com/golemcloud/golem/assets/7448613/2fc88eac-6568-4395-9fa7-405e2217c62b">

```
If the request is not application/json, then the Rib return value must be equivalent to a binary stream.
``


Previous commit supporting text: https://github.com/golemcloud/golem/blob/91f5c0122facd17180589ec19cf4344b8884ee32/golem-worker-service-base/src/worker_bridge_execution/content_type_mapper.rs#L414


I suggest @justcoon  the following: Can you discuss this in slack please?

Reply:
Or it will be helpful and easier for everyone to understand if you could make this suggestion in notion page